### DISCUSSION TITLE
Are valetudo obstacle images working?

### DISCUSSION THREAD

**Discussion by borgqueenx** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#774] [General]
Since two versions ago, valetudo supports making images of obstacles itcomes across. Im wondering if this card supports viewing those?

**Comment by PiotrMachowski** [comment]
How would you like them to be displayed? Are these images available in HA?

**Reply by borgqueenx** [reply]
I think they have to be pulled from valetudo into home assistant, and i dont know how.

**Reply by PiotrMachowski** [reply]
They have to be pulled to HA first to make it work with the card.

**Reply by borgqueenx** [reply]
So @PiotrMachowski i did some research into this myself.
The images of the robot are stored in the robot itself at /data/ai_offline_collection
The images themselfs have names that i suspect can be coordinates where the images were taken?
Here is a screenshot:
![image](https://github.com/user-attachments/assets/72347434-219d-4295-8aff-28e78f2f38f4)

I could build something myself to get the images remotely with ssh to home assistant with a script build by a AI that checks for new files and then pulls them into a home assistant pc folder. Maybe then the filenames can be checked for coordinates.

It would then probably be fun to make a slideshow of these photos to see what the robot saw, but for this map card it would be awesome to note the objects on the map and show a image when clicked. But that's beyond my scope of knowledge/coding and i also don't know how much work it would be... I am using the mqtt vacuum camera integration, that integration also probably needs editting to support this? But then maybe it can also have a service build in to pull the images so people dont have to build scripts to ssh/pull images from the robot.

I do think it would be super cool to have support for this, though. I'm curious if you like the idea of supporting that?

**Reply by PiotrMachowski** [reply]
> for this map card it would be awesome to note the objects on the map and show a image when clicked. 

That's a neat idea, but I don't think it will be possible for me to implement such functionality in the nearest future

**Reply by borgqueenx** [reply]
fair enough, thanks for reading in any case :)
